src/lib/spinel/radio_spinel.cpp

```diff
@@ -1430,7 +1430,7 @@ otError RadioSpinel::Remove(spinel_prop_key_t aKey, const char *aFormat, ...)
 
 otError RadioSpinel::WaitResponse(bool aHandleRcpTimeout)
 {
-    uint64_t end = otPlatTimeGet() + kMaxWaitTime * kUsPerMs;
+    uint64_t end = otPlatTimeGet() + static_cast<uint64_t>(kMaxWaitTime) * static_cast<uint64_t>(kUsPerMs);
```

Comment:
Thanks for finding and fixing this.

I have a suggestion for maybe a different fix. How about we define these as `constexpr` constants instead of `enum` (with `constexpr` we can be explicit about their type).

Changing:
```c++
    enum
    {
        kUsPerMs  = 1000,    ///< Microseconds per millisecond.
        kTxWaitUs = 5000000, ///< Maximum time of waiting for `TransmitDone` event, in microseconds.
    };
```
to
```c++
    static constexpr uint32_t kUsPerMs  = 1000;    ///< Microseconds per millisecond.
    static constexpr uint64_t kTxWaitUs = 5000000; ///< Maximum time of waiting for `TransmitDone` event, in microseconds.
```

Reply:
Yes, it is a gread idea to define them as `constexpr` constants. I will modify the codes. Thanks.